### ISSUE TITLE
Error with redis 5.x gem

### ISSUE THREAD

**Issue by twnaing** [sidekiq-cron/sidekiq-cron#357]
With redis 5.x gem (which use redis-client 0.7.x), Sidekiq cron prevent Sidekiq from starting due to error in redis_client.

```bash
[18] pry(main)> Sidekiq::Cron::Job.load_from_array [{name: 'unique_job', class: 'ReportMailingWorker', queue: 'mailer', cron: '1 0 * * *'}]
TypeError: Unsupported command argument type: NilClass
from /home/user/.asdf/installs/ruby/2.7.6/lib/ruby/gems/2.7.0/gems/redis-client-0.7.1/lib/redis_client/command_builder.rb:75:in `block in generate'
```

With redis 4.8.x (which does not use redis-client), Sidekiq cron loaded successfully.

Currently I version fixed the redis gem to '~> 4.8' and sidekiq cron schedules are loaded successfully.

**Comment by joshuaballoch** [comment]
I ran into this error as well - looks like pretty much the same issue:

```
TypeError: Unsupported command argument type: NilClass
/Users/users/code/clients/project/.bundle/ruby/2.7.0/gems/redis-client-0.7.4/lib/redis_client/command_builder.rb:75:in `block in generate'
```

I looked over the rest of the stacktrace and found that the error comes from an attempt in `Sidekiq::Cron::Job` to [save jobs to redis using hmset (L471)](https://github.com/ondrejbartas/sidekiq-cron/blame/master/lib/sidekiq/cron/job.rb#L471).

These were the key value pairs for my job that ran into the issue:

```
[:name,
 "job_name",
 :klass,
 "JobClass",
 :cron,
 "0 0 * * *",
 :description,
 nil,
 :args,
 "[]",
 :date_as_argument,
 false,
 :message,
 "{\"retry\":true,\"queue\":\"flood\",\"class\":\"JobClass\",\"args\":[]}",
 :status,
 "disabled",
 :active_job,
 false,
 :queue_name_prefix,
 nil,
 :queue_name_delimiter,
 nil,
 :last_enqueue_time,
 2022-03-08 05:00:01 UTC,
 :symbolize_args,
 false]
```

I tried out some class to `hmset`, and found that the issue is that this no longer accepts `nil` as an acceptable value to set.

```
[8] pry(#<Sidekiq::Cron::Job>)> conn.hmset redis_key, :queue_name_prefix, nil
TypeError: Unsupported command argument type: NilClass
```

As suggested by @twnaing, downgrading to redis 4.8.x removes the issue. Out of interest, I tried `hmset` on redis 4.8.x, and it accepts nil as an acceptable value. When reading back the value, it returns an empty string - no cast back to nil.

```
[2] pry(#<Sidekiq::Cron::Job>)> conn.hmset redis_key, :name, nil
=> "OK"
[3] pry(#<Sidekiq::Cron::Job>)> conn.hmget redis_key, :name
=> [""
```

I don't know the inner workings of Sidekiq Cron well enough but perhaps switching from `nil`s to empty strings would work? Or some fix for to cast to empty string // back to nil when reading?

**Comment by markets** [comment]
hello @joshuaballoch @twnaing 

Related discussion: https://github.com/ondrejbartas/sidekiq-cron/pull/356#issuecomment-1238717521.

**TL;DR** It seems latest Sidekiq 6.X is not "compatible" with Redis 5.X.

**Comment by javinto** [comment]
Same with 

schedule_settings= Rails.application.config_for(:schedule)
Sidekiq::Cron::Job.load_from_hash schedule_settings

Rails 6.1.7
Ruby 3.1.2
Redis 5.0.5
Sidekiq 6.5.5
Sidekiq-cron 1.8.0

2022-10-05T08:43:06.239Z pid=21328 tid=22bk WARN: TypeError: Unsupported command argument type: NilClass
2022-10-05T08:43:06.239Z pid=21328 tid=22bk WARN: /Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-client-0.9.0/lib/redis_client/command_builder.rb:37:in `block in generate'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-client-0.9.0/lib/redis_client/command_builder.rb:28:in `map!'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-client-0.9.0/lib/redis_client/command_builder.rb:28:in `generate'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-client-0.9.0/lib/redis_client.rb:211:in `call_v'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-5.0.5/lib/redis/client.rb:73:in `call_v'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-5.0.5/lib/redis.rb:167:in `block in send_command'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-5.0.5/lib/redis.rb:166:in `synchronize'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-5.0.5/lib/redis.rb:166:in `send_command'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/redis-5.0.5/lib/redis/commands/hashes.rb:51:in `hmset'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:471:in `block in save'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-6.5.5/lib/sidekiq.rb:164:in `block in redis'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/connection_pool-2.3.0/lib/connection_pool.rb:65:in `block (2 levels) in with'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `handle_interrupt'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/connection_pool-2.3.0/lib/connection_pool.rb:64:in `block in with'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `handle_interrupt'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/connection_pool-2.3.0/lib/connection_pool.rb:61:in `with'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-6.5.5/lib/sidekiq.rb:161:in `redis'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:465:in `save'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:197:in `block in load_from_array'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:195:in `each'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:195:in `load_from_array'
/Users/itsme/.rvm/gems/ruby-3.1.2@project/gems/sidekiq-cron-1.8.0/lib/sidekiq/cron/job.rb:166:in `load_from_hash'

**Comment by rromanchuk** [comment]
@joshuaballoch is there any quick patch `conn.hmset redis_key, *hash_to_redis(to_hash)`  like `to_hash.compact` etc, this gem is killing me with  `Unsupported command argument type: NilClass (TypeError)`, causing sidekiq launch failures

Doesn't matter if it's yaml or manual constructor

**Comment by joshuaballoch** [comment]
@rromanchuk I don't have a quick patch - however, if you upgrade sidekiq to 6.5.6 or later, the Sidekiq gem pins Redis to 4.8.0, so it fixes the issue. I think Sidekiq has plans to use a different gem for Redis in the future

**Comment by aforty** [comment]
Updated from sidekiq 6.5.5 to 6.5.8 which fixed the issue. Thanks @joshuaballoch!
